### PR TITLE
gh-91126: Docs and tests for slotted dataclasses with `__init_subclass__`

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-91126

----
📚 Documentation preview 📚: https://cpython-previews--123342.org.readthedocs.build/

### CODE REVIEW

**Comment by sobolevn** [comment]
New look: https://cpython-previews--123342.org.readthedocs.build/en/123342/library/dataclasses.html

Снимок экрана 2024-08-26 в 18 45 01

File: Doc/library/dataclasses.rst [hunk 1]
```diff
@@ -185,10 +185,20 @@ Module contents
...
+
+    .. warning::
+        Calling no-arg :func:`super` in dataclasses using ``slots=True``
+        will result in the following exception being raised:
+        ``TypeError: super(type, obj): obj must be an instance or subtype of type``.
+        The two-arg :func:`super` is a valid workaround.
+        See :gh:`90562` for full details.
+
+    .. warning::
+       Having a base class with :meth:`~object.__init_subclass__` with parameters
```
Comment (carljm) [inline Doc/library/dataclasses.rst]:
```suggestion
       Passing parameters to a base class :meth:`~object.__init_subclass__`
```

File: Lib/test/test_dataclasses/__init__.py [hunk 2]
```diff
@@ -3664,6 +3664,34 @@ class A(WithDictSlot): ...
         self.assertEqual(A().__dict__, {})
         A()
 
+    def test_slots_with_wrong_init_subclass(self):
```
Comment (carljm) [inline Lib/test/test_dataclasses/__init__.py:3668]: The value of adding tests for buggy behavior is not clear to me, but if we will add the tests, I'd rather have a comment making it very clear that this tested behavior is a bug that should ideally be fixed, not desired behavior.

**Review by carljm — COMMENTED** [review]
Thanks!

File: Lib/test/test_dataclasses/__init__.py (hunk 2 above)
Comment (sobolevn) [reply Lib/test/test_dataclasses/__init__.py:3668]: I've added a comment to the test, it is not very likely that we will fix this soon.

**Review by carljm — APPROVED** [review]
LGTM!

**Comment by sobolevn** [comment]
I will wait a bit for @ericvsmith's review and merge it in a couple of days. Thanks!

**Comment by sobolevn** [comment]
One week later - merging :)

**Pull request merged**